libcst/_parser/detect_config.py

```diff
@@ -114,6 +114,23 @@ def _detect_future_imports(tokens: Iterable[Token]) -> FrozenSet[str]:
     return frozenset(future_imports)
```

Comment:
I'm not 100% sure this has good tests.  Does this avoid having to tokenize twice?

Reply:
Hmm, I don't think this code is ever getting run in native mode. Will probably have to be reimplemented on the rust side in the future. As far as I can tell this is only being used to figure out the exact grammar version to be used.